test/test_mktime.c

```diff
@@ -0,0 +1,1096 @@
+static const int output_t3_1[][NCOLS];
+static const time_t output_t3_2[];
+
+static const int input_t4[][NCOLS];
+static const int output_t4_1[][NCOLS];
+static const time_t output_t4_2[];
+
+void tm_to_str(struct tm *t, char *buff){
+	sprintf(buff, "tm_sec %d, tm_min %d, tm_hour %d, tm_mday %d\n"
+			"tm_mon %d tm_year %d tm_wday %d tm_yday %d tm_isdst %d",
+			t->tm_sec, t->tm_min, t->tm_hour, t->tm_mday, t->tm_mon, t->tm_year,
+			t->tm_wday, t->tm_yday, t->tm_isdst);
+
+}
+
+struct tm create_tm(const int* input){
+	struct tm t;
+	t = (struct tm) {.tm_sec=input[0], .tm_min=input[1], .tm_hour=input[2], .tm_mday=input[3], 
+					 .tm_mon=input[4], .tm_year=input[5], .tm_wday=input[6], .tm_yday=input[7], 
+					 .tm_isdst=input[8]};
+	return t;
+}
+
+void generate_input_host(int input_length){
+	int i;
+	srand(time(NULL));
+	for (i=0; i<input_length; i++){
+	printf("{%d, %d, %d, %d, %d, %d, %d, %d, %d}", rand()%59, rand()%59, rand()%23,
+			rand()%30, rand()%11, 1980+rand()%60, 0, 0, 0);
+	if (i != input_length-1)
+		printf(",\n");
+	}
+	printf("\n");
+};
+
+void generate_output_host(const int input_vector[][NCOLS], int input_length){
+	int i;
+	struct tm t;
+	time_t timestamp;
+	printf("Printing host output data. Struct tm member values:\n");
+	for (i=0; i<input_length; i++){
+		t = create_tm(input_vector[i]);
+		timestamp = mktime(&t);
+		printf("{%d, %d, %d, %d, %d, %d, %d, %d, %d}",
+				   t.tm_sec, t.tm_min, t.tm_hour, t.tm_mday, t.tm_mon, t.tm_year,
+				   t.tm_wday, t.tm_yday, t.tm_isdst);
+		if(i != input_length-1)
+			printf(",\n");
+
+	}
+	printf("\nTimestamp values:\n{");
+	for (i=0; i<input_length; i++){
+		t = create_tm(input_vector[i]);
+		timestamp = mktime(&t);
+		printf("%lld", timestamp);
+		if(i != input_length-1)
+			printf(", ");
+	}
+	printf("}\n");
+}
+
+int compare_tm(struct tm *t1, struct tm *t2){
+	int result = 0;
+	result+= (t1->tm_sec == t2->tm_sec) ? 0 : 1;
+	result+= (t1->tm_min == t2->tm_min) ? 0 : 1;
+	result+= (t1->tm_hour == t2->tm_hour) ? 0 : 1;
+	result+= (t1->tm_mday == t2->tm_mday) ? 0 : 1;
+	result+= (t1->tm_mon == t2->tm_mon) ? 0 : 1;
+	result+= (t1->tm_year == t2->tm_year) ? 0 : 1;
+	result+= (t1->tm_wday == t2->tm_wday) ? 0 : 1;
+	result+= (t1->tm_yday == t2->tm_yday) ? 0 : 1;
+	result+= (t1->tm_isdst == t2->tm_isdst) ? 0 : 1;
+	return result;
+}
+
+int mktime_assert(const int input[], const int output[], const time_t timestamp_exp){
+	struct tm t, t_old, t_exp;
+	time_t timestamp;
+	char buff[200];
+	t = create_tm(input);
+	t_old = create_tm(input);
+	t_exp = create_tm(output);
+	timestamp = mktime(&t);
+	if (compare_tm(&t, &t_exp) || ((timestamp) != (timestamp_exp))){	
+		printf("Testcase failed\n");
+		tm_to_str(&t_old, buff);
+		printf("Original struct tm:\n%s\n", buff);
+		tm_to_str(&t, buff);
+		printf("Output struct tm:\n%s\n", buff);
+		tm_to_str(&t_exp, buff); 	
+		printf("Expected output struct tm:\n%s\n", buff);
+		printf("Timestamp: %lld, expected: %lld\n\n", timestamp, timestamp_exp);
```

Comment:
This function is too verbose. You should report success for all test or failing cases only